source/CScriptEngine.cpp

```diff
@@ -1360,23 +1396,26 @@ namespace CLEO
 				}
 				is.read(reinterpret_cast<char *>(BaseIP), length);
 
-				auto fname = strrchr(szFileName, '\\') + 1;
-				if (!fname) fname = strrchr(szFileName, '/') + 1;
-				if (fname < szFileName) fname = szFileName;
-				memcpy(Name, fname, sizeof(Name));
-				Name[7] = '\0';
-				dwChecksum = crc32(reinterpret_cast<BYTE *>(BaseIP), length);
+                dwChecksum = crc32(reinterpret_cast<BYTE*>(BaseIP), length);
+
+                // thread name from filename
+                memset(Name, '\0', sizeof(Name));
+                if(!scriptFileName.empty())
+                {
+                    auto len = min(scriptFileName.length(), sizeof(Name) - 1); // and text terminator
+                    memcpy(Name, scriptFileName.c_str(), len);

```

Comment:
one question. if script's file name is `1.cs4`, what the script name (03A4) is then? "1.cs4" ?

Reply:
yes

Reply:
can we make the name "1.cs" in this case? let .cs3 and .cs4 be virtual extensions, only to enable the compat mode, but otherwise for opcodes visible as .cs?

Reply:
so the logic for CLEO would be

- scan CLEO folder
- load all *.cs
- for all *.cs3 or cs4
  - enable  compat mode
  - store filename as .cs

any issues?

Reply:
Thread name should reflect original file type in case somebody depended on it.
Scriptname storage is introduced by CLEO5, and in my opinion it should reflect true file source, as for example it might be used to hot reload the script.

Reply:
> Thread name should reflect original file type in case somebody depended on it.

exactly. you would release the script as "1.cs" and in your code you refer to it as "1.cs". emergency.cs is another example.
but you can't change it to .cs4 without changing the logic now